filebeat/crawler/registrar.go

```diff
@@ -200,29 +200,24 @@ func (r *Registrar) getPreviousFile(newFilePath string, newFileInfo os.FileInfo)
 	return "", fmt.Errorf("No previous file found")
 }
 
-func (r *Registrar) setState(path string, state *FileState) {
+func (r *Registrar) setState(path string, state FileState) {
 	r.stateMutex.Lock()
 	defer r.stateMutex.Unlock()
 
 	r.state[path] = state
 }
 
-func (r *Registrar) getState(path string) (*FileState, bool) {
+func (r *Registrar) getStateEntry(path string) (FileState, bool) {
 	r.stateMutex.Lock()
 	defer r.stateMutex.Unlock()
 
 	state, exist := r.state[path]
 	return state, exist
 }
 
-func (r *Registrar) getStateCopy() map[string]FileState {
+func (r *Registrar) getState() map[string]FileState {
 	r.stateMutex.Lock()
 	defer r.stateMutex.Unlock()
 
-	copy := make(map[string]FileState)
-	for k, v := range r.state {
-		copy[k] = *v
-	}
-
-	return copy
+	return r.state
```

Comment:
Without making a copy of the map this method doesn't really provide any benefit/feature. Nor is there any reason to grab the lock. Maybe you still want to make a copy?